tls/conn.go

```diff
@@ -19,6 +19,8 @@ import (
 	"time"
 )
 
+var HandshakeFailureErr = &net.OpError{Op: "remote error", Err: alert(alertHandshakeFailure)}
```

Comment:
Understood. The control flow in the scan is odd: the scan loop terminates based on the receipt of an alert message, which causes the scan to always print an error. I want to catch this error and ignore it. The only reliable way to do this is to change the way alert messages are handled.

That solution isn't totally ideal, because it means that I'm overloading alertHandshakeFailure to cover (1) the expected failure at the end of the scan, when no viable parameters can be chosen and (2) any other event which would cause that alert message. In particular, once I catch that error in the scan in cfssl, I simply assign `err = nil` and move on, which effectively papers over some legitimate error cases. I've pushed that code, so you can see what I was trying to do. I'd love to talk about the right way to do this, though.

reflect.DeepEqual solves the particular comparison problem, but not the overloading problem. But it may be the best option.